test/stage3/staking.js

```diff
@@ -1,67 +1,68 @@
 const { ethers } = require('hardhat');
 const { expect } = require('chai');
 
-let owner, user1, user2, SEuro, TST, SEUROTST, INTEREST, StakingContract;
-const { etherBalances } = require('../common.js');
+let owner, user1, user2, SEuro, TST, simpleInterestRate, TGateway, StakingContract;
+const { etherBalances, getLibraryFactory } = require('../common.js');
 
 beforeEach(async () => {
   [owner, user1, user2] = await ethers.getSigners();
   const SEuroContract = await ethers.getContractFactory('SEuro');
   const ERC20Contract = await ethers.getContractFactory('DUMMY');
-  StakingContract = await ethers.getContractFactory('Staking');
+  const GatewayContract = await ethers.getContractFactory('StandardTokenGateway');
+  StakingContract = await getLibraryFactory(owner, 'Staking');
   SEuro = await SEuroContract.deploy('sEURO', 'SEUR', [owner.address]);
   TST = await ERC20Contract.deploy('TST', 'TST', 18);
+  TGateway = await GatewayContract.connect(owner).deploy(TST.address);
   TST_ADDRESS = TST.address;
   SEUR_ADDRESS = SEuro.address;
-  TOTAL_SEURO = etherBalances["ONE_MILLION"];
-  SEUROTST = 5000;
-  INTEREST = 7000; // 7%
-  HUNDRED_PC = 100000;
+  maxRewardSupply = etherBalances["ONE_MILLION"];
+  seuroPerStandardToken = 2;
+  simpleInterestRate = 5000; // 5%
+  RATE_FACTOR = 10000;
```

Comment:
`maxRewardSupply`, `seuroPerStandardToken `, `RATE_FACTOR` all unused

Reply:
linter did not catch it, maybe because it's being used in every test round (?)

Reply:
i think because they aren't declared in any scope as a variable or constant. so they just become global things that linters can't see